llvm/test/CodeGen/ARM/simplifysetcc_narrow_load.ll

```diff
@@ -27,7 +27,9 @@ define i1 @test_129_15_0(ptr %y) {
 ;
 ; CHECK-BE-LABEL: test_129_15_0:
 ; CHECK-BE:       @ %bb.0:
-; CHECK-BE-NEXT:    ldrh r0, [r0, #14]
+; CHECK-BE-NEXT:    ldr r1, [r0, #12]
```

Comment:
This is not a regression. It is supposed to be fixing a miscompile.

The 15bits masked out with the AND is at [r0, #15]  and [r0, #16]. I think the optimization to narrow to a 16-bit load is blocked due to loading 16 bits starting at [r0, #15] would be an unaligned load.